### PR TITLE
[new release] ppxlib (0.18.0)

### PR DESCRIPTION
**Pull request by NathanReb**
Standard library for ppx rewriters

- Project page: <a href="https://github.com/ocaml-ppx/ppxlib">https://github.com/ocaml-ppx/ppxlib</a>
- Documentation: <a href="https://ocaml-ppx.github.io/ppxlib/">https://ocaml-ppx.github.io/ppxlib/</a>

##### CHANGES:

- Bump ppxlib's AST to 4.11 (ocaml-ppx/ppxlib#180, @NathanReb)

### CODE REVIEW

File: packages/ppxlib/ppxlib.0.18.0/opam [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+license: "MIT"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+run-test: [
+  ["dune" "runtest" "-p" name "-j" jobs] { ocaml:version >= "4.10" }
+]
+depends: [
+  "ocaml"                   {>= "4.04.1"}
```
Comment (kit-ty-kate) [inline packages/ppxlib/ppxlib.0.18.0/opam:17]: does it support OCaml 4.12?

`ppxlib.0.17.0` fails with:
```
#=== ERROR while compiling ppxlib.0.17.0 ======================================#
# context              2.1.0~beta2 | linux/x86_64 | ocaml-variants.4.12.0+trunk | file:///home/opam/opam-repository
# path                 ~/.opam/4.12.0+trunk/.opam-switch/build/ppxlib.0.17.0
# command              ~/.opam/4.12.0+trunk/bin/dune build -p ppxlib -j 71
# exit-code            1
# env-file             ~/.opam/log/ppxlib-6-af3201.env
# output-file          ~/.opam/log/ppxlib-6-af3201.out
### output ###
# [...]
#           pp ast/location_error.pp.mli (exit 1)
# (cd _build/default && ast/pp/pp.exe 4.12.0+dev0-2020-04-22 ast/location_error.mli) > _build/default/ast/location_error.pp.mli
# Unknown OCaml version 4.12.0+dev0-2020-04-22
#           pp ast/versions.pp.mli (exit 1)
# (cd _build/default && ast/pp/pp.exe 4.12.0+dev0-2020-04-22 ast/versions.mli) > _build/default/ast/versions.pp.mli
# Unknown OCaml version 4.12.0+dev0-2020-04-22
#           pp ast/location_error.pp.ml (exit 1)
# (cd _build/default && ast/pp/pp.exe 4.12.0+dev0-2020-04-22 ast/location_error.ml) > _build/default/ast/location_error.pp.ml
# Unknown OCaml version 4.12.0+dev0-2020-04-22
#           pp ast/versions.pp.ml (exit 1)
# (cd _build/default && ast/pp/pp.exe 4.12.0+dev0-2020-04-22 ast/versions.ml) > _build/default/ast/versions.pp.ml
# Unknown OCaml version 4.12.0+dev0-2020-04-22
```
(tested with https://github.com/ocaml-ppx/ocaml-migrate-parsetree/pull/107)

Comment (kit-ty-kate) [reply packages/ppxlib/ppxlib.0.18.0/opam:17]: I got an answer to my question by looking into it myself. I opened a PR to fix this issue upstream: https://github.com/ocaml-ppx/ppxlib/pull/190

**Comment by mseri** [comment]
@NathanReb a few revdeps seem to fail with Pconst_string having an incorrect number of arguments or similar issues, like metapp, metaquot and ppx_bitstring, can you double check if they also need an upper bound before I merge the PR?

**Comment by NathanReb** [comment]
Yeah it's expected that some other packages might break, I just added the bounds to the ones I sent patches to. I'll take a look tomorrow first thing in the morning !

**Comment by NathanReb** [comment]
Where can I look at the revdeps failures? Some of the datakit builds seem to never have actually started, including ppxlib's revdeps one :/

**Comment by mseri** [comment]
Oh no. That is really unfortunate. I think the ci was restarted overnight and we lost the previous run

**Comment by mseri** [comment]
The failures are there now. I post directly the links:

1. https://ci.ocaml.org/log/saved/docker-run-e5c6d4540c2d9cc05422d324dd854f1a/6986ad2688abc57a3ae870a14c2429704fcdf36b
2. https://ci.ocaml.org/log/saved/docker-run-88776b84cadeaf9d97377a259dcd048a/1246e3f693bc9f898bd731a51bea4c71db0aa955
3. https://ci.ocaml.org/log/saved/docker-run-a2b1acab4f2403f1b2d43b4c4038cf03/163ca9525243021b9e4deef9c5452df1bec36e33
4. https://ci.ocaml.org/log/saved/docker-run-f17b94dfcd5de6d56235ef64413297f7/b8233878856d5ed8556da2753bfe7ffbfd8d4026
5. https://ci.ocaml.org/log/saved/docker-run-1d36f70ca7b681cf3853f2afcbcee52d/0b3ed34dbc3625aa67eedef3007017437c070c99
6. https://ci.ocaml.org/log/saved/docker-run-aea7dc8b21e0b51cf1abf6adba4aea52/ab152e0c15d8951aace234adb2cea349f86177ae
7. https://ci.ocaml.org/log/saved/docker-run-105125ed0e51f66f3682d789d53372eb/90a10ac3a211e02fe33a8c1e151185199688e220
8. https://ci.ocaml.org/log/saved/docker-run-d0991f71cad891cc77ff1cb377194f88/da0cf34f4bbc1bb153220a8170beefbe5e2bdfb6
9. https://ci.ocaml.org/log/saved/docker-run-39ad3c08cfe0086005eeeb50355d714c/aff6a5774473861331ef4ddc423dea9cab8224d6

**Comment by NathanReb** [comment]
I'll take care of it ASAP!

**Comment by NathanReb** [comment]
I think I fixed them all!

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**